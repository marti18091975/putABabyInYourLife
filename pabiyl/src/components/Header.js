import React, { useState, useEffect } from 'react';
import './header.css';
import userStore from "../stores/userStore";
import { loadDetails } from "../actions/detailAction";
import { NavLink } from 'react-router-dom';
import { logout } from '../actions/AuthActions';
import authStore from '../stores/authStore';


function Header(props) {
    const detailUserId = "5f4e6fc673d494545cfbadfc";
    const [name, setName] = useState();
    const [mainImage, setMainImage] = useState();
    const [detailUsers, setDetailUsers] = useState(userStore.getDetailUsers());
    const [userEmail, setUserEmail] = useState(authStore.getUserEmail());
    useEffect(() => {
        userStore.addChangeListener(onChange);


        if (detailUsers.length === 0) {
            loadDetails();
        } else if (detailUserId) {
            const detailUser = userStore.getDetailUserByEmail(userEmail);
            if (detailUser) {

                setName(detailUser.name);
                setMainImage(detailUser.mainImage);
            }
        }
        return () => userStore.removeChangeListener(onChange);
    }, [detailUsers.length, detailUserId, onChange]);

    function onChange() {
        setDetailUsers(userStore.getDetailUsers());
    }

    return (
        <section className="main__header">
            <button className="burguer-button">
                <ul className="burguer-menu">
                    <li className="menu__header"><NavLink to="/main" className="nav__header">main</NavLink></li>
                    <li className="menu__header">
                        <NavLink to="/profile" className="nav__header">profile</NavLink></li>
                    <li className="menu__header"><NavLink to="/searchFilters" className="nav__header">find users</NavLink></li>
                    <li className="menu__header"><NavLink to="/messages" className="nav__header">messages</NavLink></li>

                </ul>
            </button>
            <div className="container__header">
                <div className="title__header">Put a baby in your life</div>
                <img src="https://www.flaticon.es/svg/static/icons/svg/2444/2444603.svg" className="logo" alt=""></img>
            </div>
            <div className="container__header">
                <div className="user__name">{name}</div>
                <img src={mainImage} alt="" className="user__image"></img>
            </div>
            <button className="log-out" onClick={logout}></button>
        </section>
    )

}
export default Header;