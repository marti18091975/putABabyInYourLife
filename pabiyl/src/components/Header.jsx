import React, { useState, useEffect } from "react";
import "./header.scss";
import userStore from "../stores/userStore";
import { loadDetails } from "../actions/detailAction";
import { NavLink } from "react-router-dom";
import { logout } from "../actions/AuthActions";
import authStore from "../stores/authStore";

function Header(props) {
  const detailUserId = "5f4e6fc673d494545cfbadfc";
  const [name, setName] = useState();
  const [mainImage, setMainImage] = useState();

  const [userEmail] = useState(authStore.getUserEmail());
  const [variable, setVariable] = useState(false);
  useEffect(() => {
    userStore.addChangeListener(onChange);

    loadDetails();
    return () => userStore.removeChangeListener(onChange);
  }, [variable, userEmail, onChange]);
  function onChange() {
    if (detailUserId) {
      const detailUser = userStore.getDetailUserByEmail(userEmail);
      if (detailUser) {
        setName(detailUser.name);
        setMainImage(detailUser.mainImage);
      }
    }
  }

  setTimeout(function () {
    if (!variable) {
      setVariable(true);
    } else {
      setVariable(false);
    }
  }, 10000);

  return (
    <section className="main__header">
      <button className="burguer-button">
        <ul className="burguer-menu">
          <li className="menu__header">
            <NavLink to="/main" className="nav__header">
              main
            </NavLink>
          </li>
          <li className="menu__header">
            <NavLink to="/profile" className="nav__header">
              profile
            </NavLink>
          </li>
          <li className="menu__header">
            <NavLink to="/searchFilters" className="nav__header">
              find users
            </NavLink>
          </li>
          <li className="menu__header">
            <NavLink to="/messages" className="nav__header">
              messages
            </NavLink>
          </li>
        </ul>
      </button>
      <div className="container__header">
        <div className="title__header">Put a baby in your life</div>
        <img
          src="https://www.flaticon.es/svg/static/icons/svg/2444/2444603.svg"
          className="logo"
          alt=""
        ></img>
      </div>
      <div className="container__header">
        <div className="user__name">{name}</div>
        <img src={mainImage} alt="" className="user__image"></img>
      </div>
      <NavLink to="/">
        <button className="log-out" onClick={logout}></button>
      </NavLink>
    </section>
  );
}
export default Header;
