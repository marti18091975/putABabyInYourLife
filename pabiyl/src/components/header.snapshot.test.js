import renderer from 'react-test-renderer';
import React from 'react';
import Header from './Header';
import { BrowserRouter as Router } from 'react-router-dom';

describe('App snapshot', () => {

    const tree = renderer.create(
        <Router>
            <Header />
        </Router>
    );

    it('should match', () => {
        expect(tree.toJSON()).toMatchSnapshot();
    });
});