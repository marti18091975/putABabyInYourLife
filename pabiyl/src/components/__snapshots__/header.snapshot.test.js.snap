// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot should match 1`] = `
<section
  className="main__header"
>
  <button
    className="burguer-button"
  >
    <ul
      className="burguer-menu"
    >
      <li
        className="menu__header"
      >
        <a
          aria-current={null}
          className="nav__header"
          href="/main"
          onClick={[Function]}
        >
          main
        </a>
      </li>
      <li
        className="menu__header"
      >
        <a
          aria-current={null}
          className="nav__header"
          href="/profile"
          onClick={[Function]}
        >
          profile
        </a>
      </li>
      <li
        className="menu__header"
      >
        <a
          aria-current={null}
          className="nav__header"
          href="/searchFilters"
          onClick={[Function]}
        >
          find users
        </a>
      </li>
      <li
        className="menu__header"
      >
        <a
          aria-current={null}
          className="nav__header"
          href="/messages"
          onClick={[Function]}
        >
          messages
        </a>
      </li>
    </ul>
  </button>
  <div
    className="container__header"
  >
    <div
      className="title__header"
    >
      Put a baby in your life
    </div>
    <img
      alt=""
      className="logo"
      src="https://www.flaticon.es/svg/static/icons/svg/2444/2444603.svg"
    />
  </div>
  <div
    className="container__header"
  >
    <div
      className="user__name"
    />
    <img
      alt=""
      className="user__image"
    />
  </div>
  <a
    aria-current="page"
    className="active"
    href="/"
    onClick={[Function]}
    style={Object {}}
  >
    <button
      className="log-out"
      onClick={[Function]}
    />
  </a>
</section>
`;
